plugins/scenario/hu.bme.mit.gamma.scenario.statechart.generator/src/hu/bme/mit/gamma/scenario/statechart/generator/MonitorStatechartgenerator.xtend

```diff
@@ -0,0 +1,240 @@
+package hu.bme.mit.gamma.scenario.statechart.generator
+
+import hu.bme.mit.gamma.scenario.model.AlternativeCombinedFragment
+import hu.bme.mit.gamma.scenario.model.Delay
+import hu.bme.mit.gamma.scenario.model.InteractionDirection
+import hu.bme.mit.gamma.scenario.model.InteractionFragment
+import hu.bme.mit.gamma.scenario.model.LoopCombinedFragment
+import hu.bme.mit.gamma.scenario.model.ModalInteractionSet
+import hu.bme.mit.gamma.scenario.model.ModalityType
+import hu.bme.mit.gamma.scenario.model.NegatedModalInteraction
+import hu.bme.mit.gamma.scenario.model.OptionalCombinedFragment
+import hu.bme.mit.gamma.scenario.model.ScenarioDefinition
+import hu.bme.mit.gamma.statechart.contract.NotDefinedEventMode
+import hu.bme.mit.gamma.statechart.interface_.Component
+import hu.bme.mit.gamma.statechart.interface_.TimeUnit
+import hu.bme.mit.gamma.statechart.statechart.State
+import hu.bme.mit.gamma.statechart.statechart.StateNode
+import hu.bme.mit.gamma.statechart.statechart.TransitionPriority
+import java.math.BigInteger
+import java.util.List
+
+import static extension hu.bme.mit.gamma.statechart.derivedfeatures.StatechartModelDerivedFeatures.*
+
+class MonitorStatechartgenerator extends AbstractContractStatechartGeneration {
+
+	protected boolean skipNextinteraction = false
+	protected State componentViolation = null
+	protected State environmentViolation = null
+	protected List<Pair<StateNode, StateNode>> copyOutgoingTransitionsForOpt = newLinkedList
+
+	new(ScenarioDefinition scenario, Component component) {
+		super(scenario, component)
+	}
+
+	override execute() {
+		statechart = createSynchronousStatechartDefinition
+		intializeStatechart()
+		for (modalInteraction : scenario.chart.fragment.interactions) {
+			if (!skipNextinteraction) {
+				process(modalInteraction)
+			} else {
+				skipNextinteraction = false;
+			}
+		}
+		firstRegion.stateNodes.get(firstRegion.stateNodes.size - 1).name = scenarioStatechartUtil.accepting
```

Comment:
Is this okay? What if there are more accepting states?

Reply:
We defined the nondeterministic automaton to have only one accepting state. Multiple accepting states only come in after we determinize the automaton.

Reply:
Okay, so this is not a problem here